clients/documents/documents.go

```diff
@@ -292,73 +263,97 @@ func (c Client) patchWithRetry(ctx context.Context, id string, version int, d do
 	return
 }
 
-func (c Client) patch(ctx context.Context, id string, version int, d documents.Document) (api.Response, error) {
-	resp, err := processHttpResponse(c.client.Patch(ctx, id, version, d))
+func (c Client) patch(ctx context.Context, id string, version int, d Document) (api.Response, error) {
+	if id == "" {
+		return api.Response{}, ErrIDEmpty
+	}
```

Comment:
I'm not sure that we really need this check, but if we keep it perhaps we should wrap it to indicate that it is coming from the patch?

Reply:
I also thought about that, and I wasn't sure. The only possible way that this condition becomes true is when the POST inside Create returns success, and the body doesn't contain the ID.
I can remove it 👍